tests/e2e/workloads/amq/test_rgw_kafka_notifications.py

```diff
@@ -0,0 +1,185 @@
+        ocp_obj.wait_for_resource(
+            condition=constants.STATUS_RUNNING,
+            selector="app=kafdrop",
+            timeout=120,
+            sleep=5,
+        )
+
+        # Get the kafkadrop route
+        kafkadrop_host = self.kafkadrop_route.get().get("spec").get("host")
+
+        # # Get the kafkaendpoint
+        kafka_endpoint = f"my-cluster-kafka-bootstrap.{constants.AMQ_NAMESPACE}.svc.cluster.local:9092"
+
+        # Create bucket
+        bucketname = bucket_factory(amount=1, interface=interface)[0].name
+
+        # Get RGW credentials
+        rgw_obj = RGW()
+        rgw_endpoint, access_key, secret_key = rgw_obj.get_credentials()
+
+        # Clone notify repo
+        notify_path = clone_notify()
+
+        # Initialise to put objects
+        data = "A random string data to write on created rgw bucket"
+        obc_obj = OBC(bucketname)
+        s3_resource = boto3.resource(
+            "s3",
+            verify=retrieve_verification_mode(),
+            endpoint_url=rgw_endpoint,
+            aws_access_key_id=obc_obj.access_key_id,
+            aws_secret_access_key=obc_obj.access_key,
+        )
+        s3_client = s3_resource.meta.client
+
+        # Initialize notify command to run
+        notify_cmd = (
+            f"python {notify_path} -e {rgw_endpoint} -a {obc_obj.access_key_id} "
+            f"-s {obc_obj.access_key} -b {bucketname} -ke {kafka_endpoint} -t {self.kafka_topic.name}"
+        )
+        log.info(f"Running cmd {notify_cmd}")
+
+        # Put objects to bucket
+        assert s3_client.put_object(
+            Bucket=bucketname, Key="key-1", Body=data
+        ), "Failed: Put object: key-1"
+        run_cmd(notify_cmd)
+
+        # Validate rgw logs notification are sent
+        # No errors are seen
+        pattern = "ERROR: failed to create push endpoint"
+        rgw_pod_name = get_pod_name_by_pattern(
+            pattern="rgw", namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+        )
+        rgw_log = get_pod_logs(pod_name=rgw_pod_name[0], container="rgw")
+        assert (
+            re.search(pattern=pattern, string=rgw_log) is None
+        ), f"{pattern} msg found in the rgw logs"
+        assert s3_client.put_object(
+            Bucket=bucketname, Key="key-2", Body=data
+        ), "Failed: Put object: key-2"
+        run_cmd(notify_cmd)
+
+        # Validate message are received Kafka side using curl command
+        # A temporary way to check from Kafka side, need to check from UI
+        curl_command = (
+            f"curl -X GET {kafkadrop_host}/topic/{self.kafka_topic.name} "
+            "-H 'content-type: application/vnd.kafka.json.v2+json'"
+        )
+        json_output = run_cmd(cmd=curl_command)
+        new_string = json_output.split()
+        messages = new_string[new_string.index("messages</td>") + 1]
+        if messages.find("1") == -1:
+            raise Exception("Messages are not recieved from Kafka side")
```

Comment:
If we hit this exception it means bucket notification is not working. Right? If yes, you can add 'rgw bucket notification is not working'

Reply:
partially yes, If we observed error messages on logs that we are checking in the previous step also indicates that the rgw bucket notification is not working correctly.